crates/mdns/src/lib.rs

```diff
@@ -175,70 +438,192 @@ mod tests {
+        let networkid = "holonaute-release.holo.host";
+
+        // This is the one from which we want to see another node disapearing from its cache
+        let mut mdns = MulticastDnsBuilder::new()
+            .multicast_address("224.0.0.251")
+            .own_record(networkid, &["wss://192.168.0.88:88088?a=to-keep"])
+            .build()
+            .expect("Fail to build mDNS.");
+
+        let mut mdns_releaser = MulticastDnsBuilder::new()
+            .own_record(networkid, &["wss://192.168.0.87:88088?a=to-release"])
+            .multicast_address("224.0.0.251")
+            .build()
+            .expect("Fail to build mDNS.");
+
+        // Make itself known ion the network
+        mdns_releaser
+            .advertise()
+            .expect("Fail to advertise my existence during release test.");
+
+        // Discovering the soon leaving participant
+        mdns.discover().expect("Fail to discover.");
+
+        println!("mdns = {:#?}", &mdns.map_record);
+        // Let's check that we discovered the soon-to-be-released record
+        {
+            let records = mdns
+                .map_record
+                .get(networkid)
+                .expect("Fail to get records from the networkid");
+            assert_eq!(records.len(), 2);
+        }
+
+        // Leaving the party
+        mdns_releaser
+            .release()
+            .expect("Fail to release myself from the participants on the network.");
+
+        // Updating the cache
+        mdns.discover().expect("Fail to discover.");
+
+        println!("mdns = {:#?}", &mdns.map_record);
+        {
+            let records = mdns
+                .map_record
+                .get(networkid)
+                .expect("Fail to get records from the networkid");
+            assert_eq!(records.len(), 1);
         }
     }
+
+    #[test]
+    fn query_test() {
+        let networkid = "holonaute-query.holo.host";
+
+        let mut mdns = MulticastDnsBuilder::new()
+            .multicast_address("224.0.0.253")
+            .own_record(networkid, &["wss://192.168.0.88:88088?a=hc0"])
+            .build()
+            .expect("Fail to build mDNS.");
+
+        eprintln!("bind addr = {}", mdns.multicast_address());
+
+        let mut mdns_other = MulticastDnsBuilder::new()
+            .own_record(networkid, &["wss://192.168.0.87:88088?a=hc-other"])
+            .multicast_address("224.0.0.253")
+            .build()
+            .expect("Fail to build mDNS.");
+
+        mdns.query().expect("Fail to advertise during Query test.");
+        mdns_other
+            .discover()
+            .expect("Fail to run discovery during Query test.");
+        mdns.discover()
+            .expect("Fail to run discovery during Query test.");
+
+        eprintln!("mdns = {:#?}", &mdns.map_record);
+
+        let records = mdns
+            .map_record
+            .get(networkid)
+            .expect("Fail to get records from the networkid");
+        assert_eq!(records.len(), 2);
```

Comment:
check the record contents? You can string sort the results so they are deterministic : )